matchers/archive.go

```diff
@@ -179,3 +186,20 @@ func MachO(buf []byte) bool {
 		(buf[0] == 0xCE && buf[1] == 0xFA && buf[2] == 0xED && buf[3] == 0xFE) ||
 		(buf[0] == 0xCA && buf[1] == 0xFE && buf[2] == 0xBA && buf[3] == 0xBE))
 }
+
+// Zstandard compressed data is made of one or more frames.
+// There are two frame formats defined by Zstandard: Zstandard frames and Skippable frames.
+// See more details from https://tools.ietf.org/id/draft-kucherawy-dispatch-zstd-00.html#rfc.section.2
+func Zst(buf []byte) bool {
+	if compareBytes(buf, zstdMagic, 0) {
+		return true
+	} else {
+		// skippable frames
+		if binary.LittleEndian.Uint32(buf[:4]) & ZstdMagicSkippableMask == ZstdMagicSkippableStart {
+			userDataLength := binary.LittleEndian.Uint32(buf[4:8])
+			nextFrame := buf[8+userDataLength:]
```

Comment:
You need to check that 'buf' is large enough, you cannot trust `userDataLength`

Reply:
Thank you for your remind.